SQA_AT/src/test/java/api/StoreTest.java

```diff
@@ -0,0 +1,52 @@
+package api;
+
+import client.StoreClient;
+import io.restassured.response.Response;
+import model.Order;
+import org.testng.annotations.Test;
+
+import java.net.HttpURLConnection;
+
+import static org.testng.Assert.assertEquals;
+
+public class StoreTest {
+    @Test
+    public void getStoreInventory() {
+        StoreClient storeClient = new StoreClient();
+        Response response = storeClient.getStoreInventory();
+        assertEquals(response.statusCode(),  HttpURLConnection.HTTP_OK, "Error - incorrect status code");
+        assertEquals(response.contentType(), "application/json", "Error - incorrect content type");
+    }
+
+    @Test
+    public void createOrder() {
+        StoreClient storeClient = new StoreClient();
+        Order order = new Order();
+        Response response = storeClient.createStoreOrder(order);
+        assertEquals(response.statusCode(),  HttpURLConnection.HTTP_OK, "Error - incorrect status code");
+        assertEquals(response.contentType(), "application/json", "Error - incorrect content type");
+    }
+
+    @Test
+    public void deleteOrderById() {
+        StoreClient storeClient = new StoreClient();
+        Order order = new Order();
+        Response createOrderResponse = storeClient.createStoreOrder(order);
+        Order newOrder = createOrderResponse.as(Order.class);
+        Response deleteResponse = storeClient.deleteStoreOrder(newOrder.getId());
+        assertEquals(deleteResponse.statusCode(),  HttpURLConnection.HTTP_OK, "Error - incorrect status code");
+        Response getOrderByIdResponse = storeClient.getStoreOrderById(newOrder.getId());
+        assertEquals(getOrderByIdResponse.getStatusCode(), HttpURLConnection.HTTP_NOT_FOUND, "Error - incorrect status code");
+    }
+
+    @Test
+    public void findOrderById() {
+        StoreClient storeClient = new StoreClient();
+        Order order = new Order();
+        Response createOrderResponse = storeClient.createStoreOrder(order);
+        Order newOrder = createOrderResponse.as(Order.class);
+        Response getOrderByIdResponse = storeClient.getStoreOrderById(newOrder.getId());
+        assertEquals(getOrderByIdResponse.statusCode(),  HttpURLConnection.HTTP_OK, "Error - incorrect status code");
+        assertEquals(getOrderByIdResponse.contentType(), "application/json", "Error - incorrect content type");
+    }
+}
```

Comment:
1. Rename baseRequestSpecification -> getBaseRequestSpecification
2. Try to add properties files (can be used library 'org.aeonbits.owner')
 https://matteobaccan.github.io/owner/docs/usage/
3. createWithList ? name should say what it's actually doing 'createUsersByProvidedUsersList ' ... final List<User> why we need Final??
4. Thread.sleep(2000); - bad practice. Better to use Awaitility
https://www.vinsguru.com/selenium-webdriver-how-to-wait-for-expected-conditions-using-awaitility/

5. catch (InterruptedException e) {throw new RuntimeException(e);
 - add some message for exception
 6. Common logic should be extracted to avoid duplication (DRY principle)
 https://www.w3schools.blog/dry-dont-repeat-yourself-in-java
 
7. Create a new class Waits and add there all WebDriverWait (visibilityOfElementLocated, urlToBe etc)
   - new By.ByCssSelector("[class=location-selector__link][lang=" + language + "]") - duplication... create a separate variable	
   - Duration.ofSeconds(2)... avoid magic numbers, strings. Constants class could be created or in waits class can be defined variables
   
   8.  Extract this method to js calss ((JavascriptExecutor) driver).executeScript("arguments[0].scrollIntoView(true);", policiesList.get(0));
   
   9 . for file 'FileUtil' maybe would be more compact to use stream api:
			public static boolean isFileDownloaded(String expectedFileName) {
				File folder = new File(System.getProperty("user.dir"));
				
				return Arrays.stream(Objects.requireNonNull(folder.listFiles()))
						.filter(file -> file.isFile() && file.getName().equals(expectedFileName))
						.peek(file -> file.deleteOnExit())
						.findFirst()
						.isPresent();
			}

Note: u can use ChatGPT to do method refactoring

10. A lot of SoftAssert softAssert = new SoftAssert(); Pls define it one time on a class level

11. Steps can be defind with Fluent Interface Design Pattern
https://www.lambdatest.com/blog/fluent-interface-design-pattern/

12, Also check a Builder design pattern 
https://refactoring.guru/design-patterns/builder